core/lexicon/en/tv_widget.inc.php

```diff
@@ -7,50 +7,94 @@
-$_lang['combo_typeahead_delay_desc'] = 'The length of time in milliseconds to wait until the Type-Ahead text is displayed if Type-Ahead is enabled (defaults to 250).';
+$_lang['combo_typeahead_desc'] = 'Populate and autoselect options that match as you type after a configurable delay. (Default: No)';
+$_lang['combo_typeahead_delay'] = 'Delay';
+$_lang['combo_typeahead_delay_desc'] = 'Milliseconds before a matched option is shown. (Default: 250)';
 $_lang['date'] = 'Date';
 $_lang['date_format'] = 'Date Format';
-$_lang['date_use_current'] = 'If no value, use current date';
+$_lang['date_format_desc'] = 'Enter a format using <a href="https://www.php.net/strftime" target="_blank">php’s strftime syntax</a>.
+    <div class="example-list">Common examples include:
+        <ul>
+            <li><span class="example-input">%A %d, %B %Y</span> ('.strftime("%A %d, %B %Y").') (default format)</li>
+            <li><span class="example-input">%a, %b %e, %Y</span> ('.strftime("%a, %b %e, %Y").')</li>
+            <li><span class="example-input">%m/%d/%Y</span> ('.strftime("%m/%d/%Y").')</li>
+            <li><span class="example-input">%Y-%m-%d</span> ('.strftime("%Y-%m-%d").')</li>
+            <li><span class="example-input">%Y-%m-%d %T</span> ('.strftime("%Y-%m-%d %T").')</li>
+            <li><span class="example-input">%b %e, %Y</span> ('.strftime("%b %e, %Y").')</li>
+            <li><span class="example-input">%e %h %Y %l:%M %p</span> ('.strftime("%e %h %Y %l:%M %p").')</li>
+        </ul>
+    </div>';
+$_lang['date_use_current'] = 'Use Current Date as Fallback';
+$_lang['date_use_current_desc'] = 'When a value for this TV is not required (Allow Blank = “Yes”) and a Default Date is not specified, setting this option to “Yes” will display the current date.';
 $_lang['default'] = 'Default';
+$_lang['default_date_now'] = 'Today with Current Time';
+$_lang['default_date_today'] = 'Today (12:00 am)';
+$_lang['default_date_yesterday'] = 'Yesterday (12:00 am)';
+$_lang['default_date_tomorrow'] = 'Tomorrow (12:00 am)';
+$_lang['default_date_custom'] = 'Custom (see description below)';
 $_lang['delim'] = 'Delimiter';
 $_lang['delimiter'] = 'Delimiter';
+$_lang['delimiter_desc'] = 'One or more characters used to separate values (applicable to TVs supporting mutliple chooseable options).';
+
+# Date example helper vars
+$_format_default = 'Y-m-d';
+$_format = $this->modx->getOption('manager_date_format');
+$_seps = '-/. ';
+$_formatWithoutYear = trim(str_replace(['o','y','Y'],'', $_format_default), $_seps);
+$_formatWithoutDay = trim(str_replace(['d','D','j','l'],'', $_format_default), $_seps);
+$_formatRegexAllDays = str_replace('{..}','..', trim(str_replace(['d','D','j','l'],'{..}', $_format_default), $_seps));
+$_formatRegexShort = trim(substr($_format_default, 0, (strlen($_format_default)-1)), $_seps);
+$_timestampAheadOneMonth = strtotime('+1 month');
+$_timestampAheadAlt = strtotime('+3 months 8 days');
```

Comment:
Again, maybe get a look at this running and consider the clarity it can bring to how to fill in the field — particularly that the current required date format to use is shown. I knew it would be a point of contention but thought it was the _one_ place I saw where it could be worth doing. I always found it odd to have dates in the past as examples for this type of field, and this would keep the examples fresh and more relevant without having to change the lex regularly. Could just be me though ;-)

Reply:
I should have been a little clearer, it's a great addition. What I actually wanted to say is what Mark said in another comment about moving the logic out of the lexicon and pass it on with a placeholder.

Reply:
Got it, I think we're all on the same page now ... as mentioned related conversation, changes have been made to resolve this.